stream/src/main/java/org/apache/kafka/stream/processor/PTopologyBuilder.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.stream.processor;
+
+import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.stream.processor.internals.PTopology;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class PTopologyBuilder {
+
+    private Map<String, ProcessorClazz> processorClasses = new HashMap<>();
+    private Map<String, KafkaSource> sources = new HashMap<>();
+
+    private Map<String, List<String>> parents = new HashMap<>();
+    private Map<String, List<String>> children = new HashMap<>();
+
+    private class ProcessorClazz {
+        public Class<? extends KafkaProcessor> clazz;
+        public PConfig config;
+
+        public ProcessorClazz(Class<? extends KafkaProcessor> clazz, PConfig config) {
+            this.clazz = clazz;
+            this.config = config;
+        }
+    }
+
+    public PTopologyBuilder() {}
+
+    @SuppressWarnings("unchecked")
+    public final KafkaSource addSource(String name, Deserializer keyDeserializer, Deserializer valDeserializer, String... topics) {
+        KafkaSource source = new KafkaSource(name, keyDeserializer, valDeserializer);
+
+        for (String topic : topics) {
+            if (sources.containsKey(topic))
+                throw new IllegalArgumentException("Topic " + topic + " has already been registered by another processor.");
+
+            sources.put(name, source);
```

Comment:
(Sorry for jumping in -- I noticed this comment purely by chance.) :-)

Is Neha's question above an indication that we have a proliferation of strings as API parameters, i.e. because we are not using dedicated types we lose certain levels of (type) safety?  In the example above, the compiler is not able to tell us that we were mistakenly referring to `name` instead of `topic` because all of the params are strings.  Is it possible that users of the API might get similarly confused / hit by this, i.e. could this turn into a usability issue?  Just asking!